Goal: the CIFS_IOC_CHECKMOUNT ioctl check assumed that errors would return a
value > 0, when in fact the return value on failure is -1.  Correct this
assumption, which was allowing any user to unmount shares mounted by other
users.

Author: Steve Langasek <vorlon@debian.org>

Upstream status: submitted as bugzilla bug #4784. Will be in 3.0.28a

Index: samba-3.0.25c/source/client/umount.cifs.c
===================================================================
--- samba-3.0.25c.orig/source/client/umount.cifs.c	2007-08-26 13:09:08.419408730 +0200
+++ samba-3.0.25c/source/client/umount.cifs.c	2007-08-26 13:09:08.919384334 +0200
@@ -131,7 +131,7 @@
 		printf("user unmounting via %s is an optional feature of",thisprogram);
 		printf(" the cifs filesystem driver (cifs.ko)");
 		printf("\n\tand requires cifs.ko version 1.32 or later\n");
-	} else if (rc > 0)
+	} else if (rc != 0)
 		printf("user unmount of %s failed with %d %s\n",dir,errno,strerror(errno));
 	close(fileid);
 
